src/main/java/net/minecraftforge/items/itemhandlerobserver/IItemHandlerObserver.java

```diff
@@ -7,11 +26,29 @@
 
 public interface IItemHandlerObserver
 {
-    void onStackInserted(IItemHandler handler, int slot, @Nonnull ItemStack oldStack, @Nonnull ItemStack newStack);
+    /**
+     * @param handler the handler that was inserted in to
+     * @param slot the slot that was inserted in to
+     * @param oldStackSize the old stackSize, 0 if it was {@link ItemStack#EMPTY}
+     * @param newStack the stack that is inserted into the slot
+     */
+    void onStackInserted(IItemHandler handler, int slot, int oldStackSize, @Nonnull ItemStack newStack);
```

Comment:
How does this deal with the itemstack itself being changed? There is no way of telling, if the item itself changed. At least not without keeping a secondary cache around just to compare against.